### PR TITLE
RFC #54 - Code of Conduct

### PR DESCRIPTION
**Pull request by stephenmcgruer**
This is a rework of https://github.com/web-platform-tests/rfcs/pull/17; that RFC was opened from an organization-owned cross-fork, and @jugglinmike and I were unable to figure out how to open edit access to it for all.

There have been some discussions about the RFC in https://github.com/web-platform-tests/rfcs/issues/21; I have resolved most of the action items I came up with and added to the RFC, so I think it's ready to be discussed in this forum again.

### CODE REVIEW

**Comment by foolip** [comment]
For ease of review: https://github.com/web-platform-tests/rfcs/blob/bocoup-code-of-conduct/rfcs/code_of_conduct.md

File: rfcs/code_of_conduct.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Introduce a code of conduct which clearly defines the acceptable behavior for
+all participants in the WPT project and includes contact information for a
+group of individuals who can be trusted to enforce the policy.
+
+Publishing a code of conduct is a way "to be overt in our openness, welcoming
+all people to contribute, and pledging in return to value them as whole human
+beings and to foster an atmosphere of kindness, cooperation, and
+understanding." (via
+[contributor-covenant.org](https://www.contributor-covenant.org/)
```
Comment (Ms2ger) [inline rfcs/code_of_conduct.md:13]:
```suggestion
understanding." (via
[contributor-covenant.org](https://www.contributor-covenant.org/))
```

File: rfcs/code_of_conduct.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Details
+
+This RFC consists of two parts: ratifying the content in the [proposed
+`CODE_OF_CONDUCT.md`](https://github.com/web-platform-tests/wpt/pull/7520)
```
Comment (gsnedders) [inline rfcs/code_of_conduct.md:18]: Our existing text is based on an edited version of the Rust CoC, itself based on an old version of the Contributor Covenant. Have we considered whether we just want to base it upon the latest Contributor Covenant text? (This is purely a comment/open question, I don't know how much has changed.)

Comment (stephenmcgruer) [reply rfcs/code_of_conduct.md:18]: Assuming https://www.contributor-covenant.org/version/2/0/code_of_conduct/is the new Contributor Covenant, I took a look through. I didn't find many substantive differences (there are a few). The Contributor Covenant arguably has a more structured layout, but I found their 'Enforcement Guidelines' section actually a little confusing; I wasn't clear if they were specific examples or just general 'impact' statements.

I lean towards what we have being 'good enough', but I'm definitely open to being wrong about that. I don't have much experience in this space.

**Comment by stephenmcgruer** [comment]
Thanks to those who have reviewed so far. One thought I wanted to circulate; should we be attempting to either ask for review from, or circulate this post-acceptance to, a wider audience (i.e. the WPT community)? And if so, any ideas on how best to do so.

We could post to the 'reviewers' team on web-platform-tests, which has ~300 people, but I'm not sure how many of them filter notifications.

**Comment by foolip** [comment]
Posting to the reviewers team and an email to https://lists.w3.org/Archives/Public/public-test-infra/ seems reasonable. I expect lots of people still won't see it, but these are the best channels we have AFAIK.

**Comment by stephenmcgruer** [comment]
> Posting to the reviewers team and an email to https://lists.w3.org/Archives/Public/public-test-infra/ seems reasonable. I expect lots of people still won't see it, but these are the best channels we have AFAIK.

Did those this morning. I'm going to give this RFC another few days, aim to merge early next week unless concerns come up.

**Comment by wingo** [comment]
Hi, asked to take a look at this by my colleague @Ms2ger -- FWIW I am on the group that handles CoC reports at work.  I think this is a good effort but I would not recommend merging as-is.

 * The linked PR (https://github.com/web-platform-tests/wpt/pull/7520/files) has a dangling link to the citizen CoC, so I can't see what the standards are.
 * It does not seem to me to be a good idea to spend time writing a CoC; better to choose one that represents what you want.  The [contributor covenant 2.0](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) is pretty standard at this point and would be a good choice.
 * The summary document on this PR emphasizes responsibility to "understand both sides", which to me is not the right way to see CoC incidents.  If anything the moderators need to understand the situation and its impact on people; X said / Y said isn't the right framework for this.
 * In general a CoC is a bad place to promote desired behavior ("be kind", etc).  That happens elsewhere and in other ways.  The sole purpose of a CoC (IMO) relates to bad behavior, not "good" behavior.  If you are not convinced but interested in this argument, see chapter one in https://files.frameshiftconsulting.com/books/cocguide.pdf.

This is very much "for what it's worth" feedback; although I interact with WPT in my work on WebAssembly, I'm not an active contributor.  Good luck!

**Comment by stephenmcgruer** [comment]
@wingo - thank you for the feedback (and for keeping it constructive ❤️ ), it's very much appreciated.

At a high level, I myself have no experience with Code of Conducts, nor was I involved in the early stages of creating this one. I'm going to do some digging on what other medium-sized GitHub projects tend to do for their Code of Conducts, but I would be interested in the opinion of folks @web-platform-tests/wpt-core-team as well as the folks from Bocoup that have been involved (cc @jorydotcom @boazsender ). I believe @gsnedders also suggested to just use the Contributor Covenant (see their comment above).

We have one 'customization' which I think is important, which is the clarification on where both our Code of Conduct and another CoC would apply (usually W3C/WHATWG ones). But we could still keep that customization whilst using the Contributor Covenant I expect.

> The summary document on this PR emphasizes responsibility to "understand both sides", which to me is not the right way to see CoC incidents. If anything the moderators need to understand the situation and its impact on people; X said / Y said isn't the right framework for this.

Yes, that one is written by me and is probably badly written and/or just wrong :). I would say that I was concerned about 'witch hunts' (for lack of a better term), whilst still wanting to be very aware of and considerate to the affected party (especially given how often discrimination can creep in via the guise of 'finding proof'). I'd be happy to re-write it or drop it.

> If you are not convinced but interested in this argument, see chapter one in https://files.frameshiftconsulting.com/books/cocguide.pdf.

Thank you for the link - I'm excited to have education material to read on this subject! (As I said, I have no experience here so have very much felt like I'm muddling along trying to get this CoC business sorted :))

**Comment by jgraham** [comment]
> In general a CoC is a bad place to promote desired behavior ("be kind", etc)

I don't have an opinion either way, but that seems to at least not be universally accepted. For example the [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) which acts as a CoC for Mozilla activities does do more than set out unacceptable behaviours. I don't claim to be the right person to justify that choice, but I do know that the people working on community participation at Mozilla do have a great deal of expertise in this area.

**Comment by stephenmcgruer** [comment]
> Thank you for the link - I'm excited to have education material to read on this subject! (As I said, I have no experience here so have very much felt like I'm muddling along trying to get this CoC business sorted :))

Having reviewed this material (strongly recommended), I have removed the sentence relating to understanding both sides.

> I do know that the people working on community participation at Mozilla do have a great deal of expertise in this area.

@jgraham - if you are able to tag them into this discussion, or otherwise ask for their review, that would be great.


I'm reviewing the contributor covenant currently, with the view to lean heavier on that whilst still retaining the note about what to do when multiple CoCs apply.

**Comment by stephenmcgruer** [comment]
I have reviewed the contributor covenant code of conduct 2.0. I think it is a suitable base for us to use instead of the current ad-hoc one, and having read [the guide](https://files.frameshiftconsulting.com/books/cocguide.pdf) I think it is overall slightly better. One different thing it adds is an extra step before 'warning', called 'correction' - I would want to make sure that @nsatragno @boazsender and @jorydotcom have had the chance to review that.

I have uploaded a version based on this as https://github.com/web-platform-tests/wpt/pull/23762; for the sake of comparison the initial commit in that PR is the base contributors covenant CoC, then the second commit applies my 'WPT' diff.

Hope everyone can take a look; if there are no objections I will replace the original CoC PR content with this new one.

**Comment by stephenmcgruer** [comment]
The new code of conduct PR is now landed. Per the RFC process I'm going to start the clock 🕐  now - 7 days from today I will merge this RFC and the new code of conduct.

**Comment by guest271314** [comment]
The term "appeal" is omitted from the CoC language. 

Is the omission of an official administrative procedure to appeal fraudulent, vague, mistaken, or other erroneous actions made by  "CoC Enforcement Team/moderator" intentional, meaning there is no concept of Recourse and Remedy within this particular organization?

**Pull request merged**